pkg/R/utils.R

```diff
@@ -419,3 +419,82 @@ cleanClosure <- function(func, checkedFuncs = new.env()) {
   }
   func
 }
+
+# Common function for implmentation of zipRDD() and cartesian().
+# param
+#   x An RDD.
+#   other An RDD.
+#   operation The specified operation, "zip" or "cartesian".
+#
+# return value
+#   The result RDD.
+doZipOrCartesian <- function(x, other, operation = c("zip", "cartesian")) {
```

Comment:
Actually I like the WIP version at https://github.com/amplab-extras/SparkR-pkg/commit/02363086977cbbbd2c159c81dac5d2e9617263b4 as it gives some insight when you read `zip` as to what is going on

@concretevitamin any thoughts on this ?